sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java

```diff
@@ -77,25 +82,30 @@ public RetryPolicy() {
      * @throws NullPointerException If {@code retryOptions} is null.
      */
     public RetryPolicy(RetryOptions retryOptions) {
-        this(ImplUtils.getRetryStrategyFromOptions(retryOptions), retryOptions.getMaxRetries(),
-            retryOptions.getDelayFromHeaders());
+        this(retryOptions.getBaseDelay(), retryOptions.getMaxDelay(), retryOptions.getFixedDelay(),
```

Comment:
If retryOptions.getBaseDelay is `null` and getFixedDelay is also `null`, we should default to `DEFAULT_BASE_DELAY` and same for other values.